ais/io.py

```diff
@@ -0,0 +1,84 @@
+"""
+Objects for reading and parsing NMEA from a file.
+"""
+
+
+import codecs
+import sys
+
+import six
+
+import ais.nmea_queue
+
+
+def open(name):
+
+    """
+    Open a file containing NMEA and instantiate an instance of `NMEAFile()`.
+
+    Args:
+      name: A file path, file-like object, or '-' for stdin.
+    """
+
+    if name == '-':
+      fobj = sys.stdin
+    elif isinstance(name, six.string_types):
+      fobj = codecs.open(name, encoding='utf-8')
+    elif hasattr(name, 'close') and (
+                hasattr(name, 'next') or hasattr(name, '__next__')):
+      fobj = name
+    else:
```

Comment:
@schwehr The mode `rU` is useful for dealing with a file that has multiple or non-standard newline characters, which I had to use recently on a CSV.  It's probably far less likely that we will encounter AIS data with this issue, but its easy to support.